File: frontend/thewater/lib/models/fish_card.dart
class FishCard {
  final int cardId;
  final int userId;
  final int pointId;
  final int fishId;
  final int? aquariumId;
  final String fishName;
  final int fishSize;
  final String collectDate;
  final int sky;
  final int tw;
  final int tide;
  final String comment;
  final bool isDeleted;
  final String imgUrl;

  FishCard({
    required this.cardId,
    required this.userId,
    required this.pointId,
    required this.fishId,
    this.aquariumId,
    required this.fishName,
    required this.fishSize,
    required this.collectDate,
    required this.sky,
    required this.tw,
    required this.tide,
    required this.comment,
    required this.isDeleted,
    required this.imgUrl,
  });

  factory FishCard.fromJson(Map<String, dynamic> json) {
    return FishCard(
      cardId: json['card_id'],
      userId: json['uer_id'],
      pointId: json['point_id'],
      fishId: json['fish_id'],
      aquariumId: json['aquarium_id'],
      fishName: json['fish_name'],
      fishSize: json['fish_size'],
      collectDate: json['collect_date'],
      sky: json['sky'],
      tw: json['tw'],
      tide: json['tide'],
      comment: json['comment'],
      isDeleted: json['is_deleted'],
      imgUrl: json['img_url'],
    );
  }
}

File: BE/thewater/src/main/java/fishermanjoeandchildren/thewater/db/entity/FishCard.java
package fishermanjoeandchildren.thewater.db.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name="fish_card")
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FishCard extends Common{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "fish_card_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name="member_id")
    private Member member;

    @ManyToOne
    @JoinColumn(name = "aquarium_id", nullable = false)
    private Aquarium aquarium;

    @ManyToOne
    @JoinColumn(name="fish_id", nullable = false)
    private Fish fish;

    @Column(name = "fish_name", length = 50)
    private String fishName;

    @Column(name = "fish_size", nullable = false)
    private Double fishSize;

    @Column(name = "collect_date", nullable = false)
    private LocalDate collectDate;

    //날씨
    @Column(nullable = false)
    private Integer sky;

    //기온
    @Column
    private Double temperature;

    //수온
    @Column(name = "water_temp", nullable = false)
    private Double waterTemperature;

    //조류
    @Column
    private Integer tide;

    @Column(name = "user_comment", nullable = false)
    private String comment;

    @Column(name="has_deleted")
    private Boolean hasDeleted=false;

    @Column(name="card_img", nullable = false)
    private String cardImg;

    @Column(name = "has_visible")
    private Boolean hasVisible=false;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;


    public boolean changeFishVisible(){
        hasVisible = !hasVisible;
        return hasVisible;
    }
}
